apps/explorer/src/ui/DisplayBox.tsx

```diff
@@ -1,21 +1,17 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { useState, useCallback, useEffect } from 'react';
+import { Dialog, Transition } from '@headlessui/react';
```

Comment:
I'd rename this, `DisplayBox` is not obvious at all. I also think this needs to be broken down into a couple base components, and composed here vs having all of the logic defined here. A lot of what's defined here is use-case specific as opposed to generalized.

Reply:
What name would you like for this component?